astropy/timeseries/sampled.py

```diff
@@ -141,37 +141,61 @@ def time(self):
         """
         return self['time']
 
-    def fold(self, period=None, midpoint_epoch=None):
+    @deprecated_renamed_argument('midpoint_epoch', 'epoch_time', '4.0')
+    def fold(self, period=None, epoch_time=None, epoch_phase=0, normalize_phase=True,
+             phase_length=1):
         """
         Return a new `~astropy.timeseries.TimeSeries` folded with a period and
-        midpoint epoch.
+        epoch.
 
         Parameters
         ----------
         period : `~astropy.units.Quantity`
             The period to use for folding
-        midpoint_epoch : `~astropy.time.Time`
+        epoch_time : `~astropy.time.Time`
             The time to use as the midpoint epoch, at which the relative
-            time offset will be 0. Defaults to the first time in the time
-            series.
+            time offset / phase will be ``epoch_phase``.
+            Defaults to the first time in the time series.
+        epoch_phase : float
+            Phase of the midpoint epoch. Defaults to 0.
```

Comment:
The wording is confusing here - if it's the phase of the midpoint epoch, then why rename midpoint epoch to epoch time? I think this needs to be clearer on what impact this has on the final result.

Reply:
Also should we default to 0.5? I think we'd decided originally to make the parameter the midpoint epoch not the start epoch and I think we must have had good reasons so I'd suggest defaulting to the same? Also you should make it clear that this is a normalized phase.